agent/consul/kvs_endpoint_test.go

```diff
@@ -287,6 +287,9 @@ func TestKVSEndpoint_List(t *testing.T) {
 			t.Fatalf("bad: %v", d)
 		}
 	}
+	if dirent.QueryMeta.ResultsFilteredByACLs {
+		t.Fatal("ResultsFilteredByACLs should not be true")
+	}
```

Comment:
I think this suffices as a test but I was curious if we could actually test for the presence of the header somehow as well.

But this may actually be covered, generally, thru https://github.com/hashicorp/consul/pull/11569 ; need to double check

Reply:
Ah yeah, it is covered more generally by the `setMeta` tests in that PR, but we certainly could have better coverage by testing it on the individual HTTP handlers themselves.

On balance, I decided that, aside from the HTTP-only endpoints (in #11610), testing at the RPC level is an acceptable trade-off of value vs maintainance burden.

What do you think? 🙇🏻‍♂️